package/network/services/dnsmasq/files/dnsmasq.init

```diff
@@ -349,6 +350,10 @@ dhcp_host_add() {
 		duids="id:${duid// */}"
 	fi
 
+	if [ -n "$clientid" ]; then
+		clientid="id:${clientid},"
+	fi
+
 	if [ -z "$macs" ] && [ -z "$duids" ]; then
```

Comment:
In case macs and duids is empty macs is set to name so a host can be identified:
Now as clientid also allows to identify a host it should also be checked if clientid is empty before macs is assigned the name value

Reply:
agree, fixed